File: java/common-support/src/main/java/net/multi/language/common/message/command/order/CreateOrderCommand.java
package net.multi.language.common.message.command.order;

import net.multi.language.common.marker.message.Channel;
import net.multi.language.common.marker.message.MessageDetails;
import net.multi.language.common.marker.message.Service;
import net.multi.language.common.message.AbstractCommand;

import java.util.Objects;

@MessageDetails(
        publisher = Service.ORDER_SERVICE,
        subscribers = Service.ORDER_SERVICE,
        channel = Channel.CREATE_ORDER
)
public class CreateOrderCommand extends AbstractCommand {

    private static final String NAME = "CreateOrder";

    private final String productName;
    private final double orderTotal;
    private final long customerId;

    public CreateOrderCommand(final long customerId, final long messageId, final String productName, final double orderTotal) {
        super(messageId);
        this.customerId = customerId;
        this.productName = productName;
        this.orderTotal = orderTotal;
    }

    public long getCustomerId() {
        return customerId;
    }

    public String getProductName() {
        return productName;
    }

    public double getOrderTotal() {
        return orderTotal;
    }

    @Override
    public String getName() {
        return NAME;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        CreateOrderCommand that = (CreateOrderCommand) o;
        return Double.compare(that.orderTotal, orderTotal) == 0 &&
                customerId == that.customerId &&
                Objects.equals(productName, that.productName);
    }

    @Override
    public int hashCode() {
        return Objects.hash(productName, orderTotal, customerId);
    }
}

File: python/src/common/message/command/order/create_order_command.py
from typing import ClassVar

from common.message.abstract_command import AbstractCommand


class CreateOrderCommand(AbstractCommand):
    NAME: ClassVar[str] = "CreateOrder"

    customer_id: int
    product_name: str
    order_total: float

    def get_name(self) -> str:
        return self.NAME

    def __eq__(self, other) -> bool:
        if not isinstance(other, CreateOrderCommand):
            return False
        return (
            self.customer_id == other.customer_id
            and self.product_name == other.product_name
            and self.order_total == other.order_total
        )

    def __hash__(self) -> int:
        return hash((self.product_name, self.order_total, self.customer_id))
